Update references (#4555)
circleci/circleci-docs: jekyll/_cci2/reusing-config.md

It is also possible to allow an orb to define the executor used by all of its commands. This allows users to execute the commands of that orb in the execution environment defined by the orb's author.

### Example of Using an Executor Declared in `config.yml` with Matrix Jobs.
{:.no_toc}

> The following example declares a docker based executor which contains a node image with the tag portion of the image string parameterized with a `version` parameter. Further, in our `test` job, we similarly included a version parameter so that it may be "passed through" the job into the executor when we call it from a workflow.

Reviewer: ```suggestion
The following example declares a Docker executor with a node image. The tag portion of the image string is parameterized with a `version` parameter. A `version` parameter is also included in our `test` job so that it can be passed through the job into the executor when the job is called from a workflow.
```

> When calling the `test` job from the workflow, we can take advantage of [matrix jobs](https://circleci.com/docs/2.0/configuration-reference/#matrix-requires-version-21) to run your job multiple times in parallel with a different set of parameters. In this instance, we may test our node application against many version of Node.js

Reviewer: ```suggestion
When calling the `test` job in the workflow, [matrix jobs](https://circleci.com/docs/2.0/configuration-reference/#matrix-requires-version-21) are used to run the job multiple times, concurrently, with a different set of parameters for each. The node application is tested against many versions of Node.js
```

**Note:** The `environment` variable maps are additive. If an `executor` has one of the same `environment` variables as the `job`, the value in the job will be used.

```yaml
version: 2.1
executors:
  node:
    docker:
      - image: cimg/node:lts
    environment:
     ENV: ci
jobs:
  build:
    docker:
      - image: cimg/base:stable
    # The test executor below will be overwritten by the more explicit "docker" executor. Any env vars will be added.
    executor: node
    steps:
      - run: echo "Node will not be installed."
```

The above config would resolve to the following:

```yaml
version: 2.1
jobs:
  build:
    docker:
      - image: cimg/base:stable
    environment:
     ENV: ci       # From executor.
    steps:
      - run: echo "Node will not be installed."
```

## Using the `parameters` Declaration

Parameters are declared by name under a job, command, or executor. The immediate children of the `parameters` key are a set of keys in a map.

> The following example a command named `greeting` is designed with a single parameter named `to`. the `to` parameter is used within the steps to echo _"hello"_ back to the user.

Reviewer: ```suggestion
In the following example, a command named `greeting` is designed with a single parameter named `to`. The `to` parameter is used within the steps to echo _Hello_ back to the user.
```
